### DISCUSSION TITLE
Should copy from localhost over insecure protocols be allowed?

### DISCUSSION THREAD

**Discussion by ClSlaid** [databendlabs/databend#7812] [Q&A] [answered]
**Summary**
Currently, `Databend` disallows copying from insecure data sources. When judging whether a connection is secure, `Databend` simply checks if it's using secure protocols like `HTTPS`.

The [Secure Contexts](https://developer.mozilla.org/en-US/docs/Web/Security/Secure_Contexts#when_is_a_context_considered_secure) of web browsers trusts locally delivered user contents:

> Locally-delivered resources such as those with `http://127.0.0.1` URLs, `http://localhost` and `http://*.localhost` URLs (e.g. `http://dev.whatever.localhost/`), and `file://` URLs are also considered to have been delivered securely.

## Advantages

Allowing users to copy from `localhost` should make copying files more effortless, and debugging easier since developers can simply establish those (mocked) services locally.

## Disadvantages

Databend Cloud maybe should not allow this feature since it may make users access "local" files, so there should be a flag to enable or disallow insecure connection between localhost.

**Comment by Xuanwo** [comment, marked as answer]
> Currently, Databend disallows copying from insecure data sources

Start `databend-query` with `allow_insecure`.

**Comment by youngsofun** [comment]
it is useful when I need to copy a big file (like 60GB) locally for test.
